src/pages/api/auth/login.js

```diff
@@ -0,0 +1,50 @@
+import { PrismaClient } from "@prisma/client";
+import { generateToken } from "../../../utils/generateToken";
+import nookies from "nookies";
+import bcrypt from "bcrypt";
+
+const prisma = new PrismaClient();
+
+export default async function handler(req, res) {
+  if (req.method === "POST") {
+    const { email, password } = req.body;
+
+    if (!email || !password) {
+      return res.status(400).json({ error: "Email and password are required" });
+    }
+
+    try {
+      const user = await prisma.users.findUnique({
+        where: { email },
+      });
+
+      if (!user) {
+        return res.status(401).json({ error: "Invalid email or password" });
+      }
+      const isPasswordValid = await bcrypt.compare(password, user.password);
+      if (!isPasswordValid) {
+        return res.status(401).json({ error: "Invalid email or password" });
+      }
+
+      const token = generateToken(user);
```

Comment:
don't you need to save the token somewhere?

Reply:
I am storing it in the Cookie, isn't that enough?  This is the way I usually do it, storing the token in the cookie and then later with each request from the user that has logged in, I will send the token and they decode it so I know which user is this. This is what I know, but is there other things that I don't know? I would like to learn.

@MohammedAlabd